### ISSUE TITLE
np.exp() and np.log() overflow when using Softmax/Sigmoid/CrossEntropyLoss

### ISSUE THREAD

**Issue by rrmina** [rrmina/eureka#1]
![image](https://user-images.githubusercontent.com/7788765/55301982-c387b800-5471-11e9-84fe-fa54ee5af66b.png)

Files and classes affected:

1. activation.py
   - Sigmoid
   - Softmax

2. losses.py
   - CrossEntropyLoss

**Comment by rrmina** [comment]
So I have encountered this problem again while using eureka in training a Fully-connected layer over CIFAR-10 dataset. I found out that:
1. This problem is **not** caused by RMSProp.
2. This is actually a **numerical stability problem**.
3. This might be caused by _bad initialization_, which resulted to **gradient explosion**!
4. This is implicitly solved by using BatchNorm.

Possible Solutions:

- [ ] Penalize weight values of wider linear layers
- [ ] Traceback gradient flow in a Linear layer.
